umpost/um2netcdf.py

```diff
@@ -380,38 +380,16 @@ def cubewrite(cube, sman, compression, use64bit, verbose):
         if coord.points.dtype == np.int64:
             coord.points = coord.points.astype(np.int32)
 
-    try:
-        # If time is a dimension but not a coordinate dimension, coord_dims('time') returns empty tuple
-        if tdim := cube.coord_dims('time'):
-            # For fields with a pseudo-level, time may not be the first dimension
-            if tdim != (0,):
-                tdim = tdim[0]
-                neworder = list(range(cube.ndim))
-                neworder.remove(tdim)
-                neworder.insert(0, tdim)
+    cube, unlimited_dimension = fix_time_coord(cube, verbose)
 
-                if verbose > 1:
-                    print("Incorrect dimension order", cube)
-                    print("Transpose to", neworder)
-
-                cube.transpose(neworder)
-
-            sman.write(cube,
-                       zlib=True,
-                       complevel=compression,
-                       unlimited_dimensions=['time'],
-                       fill_value=fill_value)
-        else:
-            tmp = iris.util.new_axis(cube, cube.coord('time'))
-            sman.write(tmp,
-                       zlib=True,
-                       complevel=compression,
-                       unlimited_dimensions=['time'],
-                       fill_value=fill_value)
-
-    except iris.exceptions.CoordinateNotFoundError:
-        # No time dimension (probably ancillary file)
-        sman.write(cube, zlib=True, complevel=compression, fill_value=fill_value)
+    # TODO: refactor & move to end of process()
+    # TODO: refactor cubewrite() to return (cube, unlimited dims, fill value)
+    #       then move above steps into process() / remove cubewrite()
+    sman.write(cube,
+               zlib=True,
+               complevel=compression,
+               unlimited_dimensions=unlimited_dimension,
+               fill_value=fill_value)
 
```

Comment:
```suggestion
    cube, unlimited_dimensions = fix_time_coord(cube, verbose)

    # TODO: refactor & move to end of process()
    # TODO: refactor cubewrite() to return (cube, unlimited dims, fill value)
    #       then move above steps into process() / remove cubewrite()
    sman.write(cube,
               zlib=True,
               complevel=compression,
               unlimited_dimensions=unlimited_dimensions,
               fill_value=fill_value)

```

Small change, but I suggest using the plural. It's an array, and in `fix_time_coord` it is referred to as a plural too. Just less confusing.

Reply:
GitHub seems to have done something quite odd with this (including the original code). It is a small change: just making the variable name a plural for consistency.

Reply:
That's made it to e31c5425.